packages/deployments/contracts/test/connext.spec.ts

```diff
@@ -1042,4 +1018,321 @@ describe("Connext", () => {
   });
+
+  describe("multipath", () => {
+    const params = {
+      to: user.address,
+      callData: "0x",
+      originDomain,
+      destinationDomain,
+    };
+    const amount = 1000;
+    let message: any;
+    let nonce: any;
+    let transferId: any;
+    let bridgedAmount: any;
+    let reconciledTopics: any;
+
+    beforeEach(async () => {
+      await originTm.setupRouter(router2.address, router2.address, router2.address);
+      await originTm.setupRouter(router3.address, router3.address, router3.address);
+      await destinationTm.setupRouter(router2.address, router2.address, router2.address);
+      await destinationTm.setupRouter(router3.address, router3.address, router3.address);
+      // Mint to routers
+      await local.mint(router2.address, parseEther("20")).then((r) => r.wait());
+      await local.mint(router3.address, parseEther("20")).then((r) => r.wait());
+
+      // Add routers liquidity
+      await local.connect(router).approve(destinationTm.address, parseEther("100000"));
+      await destinationTm.connect(router).addLiquidity(parseEther("0.1"), local.address);
+
+      await local.connect(router2).approve(destinationTm.address, parseEther("100000"));
+      await destinationTm.connect(router2).addLiquidity(parseEther("0.1"), local.address);
+
+      await local.connect(router3).approve(destinationTm.address, parseEther("100000"));
+      await destinationTm.connect(router3).addLiquidity(parseEther("0.1"), local.address);
+
+      // Setup stable swap for adopted => canonical on origin
+      await stableSwap.connect(admin).setupPool(originAdopted.address, canonical.address, SEED, SEED);
+
+      // Setup stable swap for local => adopted on dest
+      await stableSwap.connect(admin).setupPool(destinationAdopted.address, local.address, SEED * 2, SEED * 2);
+
+      // Approve user
+      await originAdopted.connect(user).approve(originTm.address, parseEther("100000"));
+
+      // Prepare from the user
+      const transactingAssetId = originAdopted.address;
+      const prepare = await originTm.connect(user).xcall({ params, transactingAssetId, amount });
+      const prepareReceipt = await prepare.wait();
+
+      // Get the message + id from the events
+      const topics = originBridge.filters.Send().topics as string[];
+      const bridgeEvent = originBridge.interface.parseLog(
+        prepareReceipt.logs.find((l) => l.topics.includes(topics[0]))!,
+      );
+      message = (bridgeEvent!.args as any).message;
+      bridgedAmount = (bridgeEvent!.args as any).amount;
+
+      const originTmEvent = (await originTm.queryFilter(originTm.filters.XCalled())).find(
+        (a) => a.blockNumber === prepareReceipt.blockNumber,
+      );
+      nonce = (originTmEvent!.args as any).nonce;
+      transferId = (originTmEvent!.args as any).transferId;
+
+      reconciledTopics = destinationTm.filters.Reconciled().topics as string[];
+    });
+
+    it("should work with one router", async () => {
```

Comment:
This is the same test as is run in the other tests, right?

Reply:
Yes, just with different number of routers. Made a refactor here https://github.com/connext/nxtp/pull/916/commits/0e74b944169e3883c7cf3c0985fde311d160b1e8